spec/models/miq_worker_spec.rb

```diff
@@ -267,6 +267,35 @@ def check_has_required_role(worker_role_names, expected_result)
                .worker_settings(:config => settings)[:memory_threshold]
       expect(actual).to eq(1.terabyte)
     end
+
+    context "with encrypted values" do
+      let(:settings) do
+        {
+          :workers => {
+            :worker_base => {
+              :defaults          => {:memory_threshold => "100.megabytes"},
+              :queue_worker_base => {
+                :defaults           => {:memory_threshold => "300.megabytes"},
+                :ems_refresh_worker => {
+                  :defaults                  => {:memory_threshold => "500.megabytes"},
+                  :ems_refresh_worker_amazon => {
+                    :memory_threshold => "700.megabytes",
+                    :proxy_password   => ManageIQ::Password.encrypt("password")
+                  }
+                }
+              }
+            }
+          },
+          :ems     => {:ems_amazon => {}}
+        }
+      end
+
+      it "decrypts values" do
+        expect(
+          ManageIQ::Providers::Amazon::CloudManager::RefreshWorker.worker_settings[:proxy_password]
+        ).to eq("password")
+      end
+    end
```

Comment:
please add check for `700.megabytes` blowing up the number

Reply:
We have specs for handling `"700.megabytes"` above, the only new code here is to handle decryption.  Can you check the above specs in this context and see if your concern is already covered there?